lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1059,7 +1068,7 @@ def _stop_opensearch(self, *, restart=False) -> None:
                 # and re-using storage
                 if len(nodes) > 1:
                     # 1. Add current node to the voting + alloc exclusions
-                    self.opensearch_exclusions.add_current(restart=restart)
+                    self.opensearch_exclusions.add_current()
```

Comment:
@Mehdi-Bendriss agreed for allocation, but not for exclusion. I will update it, but I'd rather have the `restart` boolean dealt on the `_stop` / `_start` methods, e.g.:
```
self.opensearch_exclusions.add_current(voting=True, allocation=not restart)
```

Reply:
My main concern is: what if we were resetting a single node of 3, and we fail to bring it back online... Then, we may end up with a cluster error because the 3->2 scaling only works if we carefully exclude the unit from the voting pool before stopping.

If we fail bringing up or even take longer than the default wait time ( `30s`), then we will break the cluster in this case.